srcts/src/components/_shinyResizeObserver.ts

```diff
@@ -0,0 +1,138 @@
+   * resize since all the widgets will return to their original size (and thus,
+   * Shiny thinks there isn't any resizing to do). The resize observer works
+   * around this by ensuring that the output is resized whenever its container
+   * size changes.
+   * @constructor
+   */
+  constructor() {
+    this.resizeObserverEntries = [];
+    this.resizeObserver = new ResizeObserver((entries) => {
+      const resizeEvent = new Event("resize");
+      window.dispatchEvent(resizeEvent);
+
+      const resized = [] as HTMLElement[];
+
+      for (const entry of entries) {
+        if (!(entry.target instanceof HTMLElement)) continue;
+        if (!entry.target.querySelector(".shiny-bound-output")) continue;
+
+        entry.target.querySelectorAll(".shiny-bound-output").forEach((el) => {
+          if (!(el instanceof HTMLElement)) return;
+          if (resized.includes(el)) return;
+
+          const { binding, onResize } = $(el).data("shinyOutputBinding");
+          if (!binding || !binding.resize) return;
+
+          // if this output is owned by another observer, skip it
+          const owner = (el as any).shinyResizeObserver;
+          if (owner && owner !== this) return;
+
+          // trigger immediate resizing of outputs with a resize method
+          onResize(el);
+          // only once per output and resize event
+          resized.push(el);
+          // mark this output as owned by this shinyResizeObserver instance
+          (el as any).shinyResizeObserver = this;
+        });
+
+        // set plot images to 100% width/height temporarily during the transition
+        entry.target
+          .querySelectorAll('.shiny-plot-output img:not([width="100%"])')
+          .forEach((el) => {
+            if (!(el instanceof HTMLImageElement)) return;
+            el.setAttribute("width", "100%");
+            el.setAttribute("height", "100%");
+          });
+      }
+    });
+  }
+
+  /**
+   * Observe an element for size changes.
+   * @param {HTMLElement} el - The element to observe.
+   */
+  observe(el: HTMLElement): void {
+    this.resizeObserver.observe(el);
+    this.resizeObserverEntries.push(el);
+  }
+
+  /**
+   * This method checks that we're not continuing to watch elements that no
+   * longer exist in the DOM. If any are found, we stop observing them and
+   * remove them from our array of observed elements.
+   *
+   * @private
+   * @static
+   */
+  flush(): void {
+    if (this.resizeObserverEntries.length == 0) return;
+
+    // Find the elements in the ResizeObserver that no longer exist in the DOM
+    const missing = this.resizeObserverEntries.filter((el) => {
+      return !document.body.contains(el);
+    });
+
+    // Remove the non-existent elements from the ResizeObserver
+    missing.forEach((el) => {
+      this.resizeObserver.unobserve(el);
+      const idxEl = this.resizeObserverEntries.indexOf(el);
+      this.resizeObserverEntries.splice(idxEl, 1);
```

Comment:
Maybe wrap this up in a `unobserve()`?

Reply:
Done in 48ff82f9202d398a0147bb27e162a126fec0afd6